src/iterrr.nim

```diff
@@ -91,6 +93,11 @@ func toIterrrPack(calls: seq[NimNode]): IterrrPack =
         expr =
           if matchInfix(firstParam, "=>"):
             firstParam[InfixRightSide]
+          elif firstParam.kind == nnkStmtList and matchInfix(firstParam[0], "=>"):
+            if firstParam[0].len < 4:
```

Comment:
what is this check exactly?

Reply:
```nim
if firstParam[0].len < 4:
```

Reply:
It distinguishes between nested and non-nested expressions. The next version of the PR will use the infixBody constant from macroplus instead of the numeric literal ‘4’